diesel/src/query_builder/bind_collector.rs

```diff
@@ -46,6 +46,23 @@ pub trait BindCollector<'a, DB: TypeMetadata>: Sized {
     }
 }
 
+/// Bind data from a BindCollector, that can be moved across threads
+pub trait MoveableBindData: Send + 'static {}
+
+/// A movable version of the bind collector which allows it to be extracted, moved and refilled.
+///
+/// This is mostly useful in async context where bind data needs to be moved across threads.
+pub trait MoveableBindCollector<DB: TypeMetadata> {
+    /// The movable bind data of this bind collector
+    type BindData: MoveableBindData;
+
+    /// Builds a movable version of the bind collector
+    fn moveable(&self) -> Self::BindData;
+
+    /// Refill the bind collector with its bind data
+    fn fill_bind_data(&mut self, from: &Self::BindData);
```

Comment:
Maybe `append_bind_data` is a better name?

Reply:
Sure, renamed